overlord/snapstate/snapmgr.go

```diff
@@ -289,7 +289,18 @@ func (m *SnapManager) doClearSnapData(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	return m.backend.RemoveSnapData(info)
+	if err = m.backend.RemoveSnapData(info); err != nil {
+		return err
+	}
+
+	// Only remove data shared between versions if this is the last version
```

Comment:
That sounds like new functionality, though-- do we currently support clearing data out from under an installed snap? As part of that feature I expected we'd also clear the shared data.